js/ui/file_manager/ui.file_manager.file_uploader.js

```diff
@@ -0,0 +1,380 @@
+import $ from "../../core/renderer";
+import { extend } from "../../core/utils/extend";
+import eventsEngine from "../../events/core/events_engine";
+import { Deferred, when } from "../../core/utils/deferred";
+
+import Widget from "../widget/ui.widget";
+import Button from "../button";
+import ProgressBar from "../progress_bar";
+import Popup from "../popup";
+
+const FILE_MANAGER_FILE_UPLOADER_CLASS = "dx-filemanager-fileuploader";
+const FILE_MANAGER_FILE_UPLOADER_FILE_INPUT_CLASS = FILE_MANAGER_FILE_UPLOADER_CLASS + "-fileinput";
+
+const FILE_MANAGER_PROGRESS_PANEL = "dx-filemanager-progresspanel";
+
+const FILE_MANAGER_PROGRESS_BOX = "dx-filemanager-progressbox";
+const FILE_MANAGER_PROGRESS_BOX_TITLE = FILE_MANAGER_PROGRESS_BOX + "-title";
+const FILE_MANAGER_PROGRESS_BOX_PROGRESS_BAR = FILE_MANAGER_PROGRESS_BOX + "-progressbar";
+const FILE_MANAGER_PROGRESS_BOX_CANCEL_BUTTON = FILE_MANAGER_PROGRESS_BOX + "-cancel-button";
+
+class FileManagerFileUploader extends Widget {
+
+    _initMarkup() {
+        this._progressPanel = this._createComponent($("<div>"), FileManagerUploadProgressPanel, {});
+
+        this.$element()
+            .addClass(FILE_MANAGER_FILE_UPLOADER_CLASS)
+            .append(this._progressPanel.$element());
+
+        this._renderFileInput();
+
+        super._initMarkup();
+    }
+
+    _renderFileInput() {
+        this._$fileInput = $("<input>")
+            .attr("type", "file")
+            .prop({
+                multiple: "multiple",
+                tabIndex: -1
+            })
+            .addClass(FILE_MANAGER_FILE_UPLOADER_FILE_INPUT_CLASS);
+
+        eventsEngine.on(this._$fileInput, "change", this._onFileInputChange.bind(this));
+        eventsEngine.on(this._$fileInput, "click", e => {
+            e.stopPropagation();
+            return true;
+        });
+
+        this.$element().append(this._$fileInput);
+    }
+
+    _onFileInputChange() {
+        const files = this._$fileInput.prop("files");
+        if(files.length === 0) {
+            return;
+        }
+
+        eventsEngine.off(this._$fileInput, "change");
```

Comment:
You can add a namespace to the events that you attach to the fileInput and remove event listeners from target element by specific namespace.

```suggestion
        eventsEngine.off(this._$fileInput, FILEMANAGER_NAMESPACE);
```

```js
const clickEvent = eventUtils.addNamespace("click", FILEMANAGER_NAMESPACE);
```